plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -128,6 +139,31 @@ public void widgetSelected(SelectionEvent e) {
     Dialog.applyDialogFont(contents);
     return contents;
   }
+  
+  private void updateSelectedVersion() {
+    String version = Messages.getString("UnknownVersion"); //$NON-NLS-1$
+    if (!CloudSdkManager.isManagedSdkFeatureEnabled()
+        || (useLocalSdk != null && useLocalSdk.getSelection())) {
+      Path path = Paths.get(sdkLocation.getStringValue());
+      try {
+        version = new CloudSdk.Builder().sdkPath(path).build().getVersion().toString();
+      } catch (CloudSdkVersionFileException ex) {
+        version = Messages.getString("NoSdkFound"); //$NON-NLS-1$
+      }
+    } else if (CloudSdkManager.isManagedSdkFeatureEnabled()) {
```

Comment:
Did you mean `CloudSdkManager.isAutoManaging()`? `isManagedSdkFeatureEnabled()` is our temporary debug flag.

Reply:
Can't check `isAutoManaging()` as this dialog is changing that result.  But the logic here is becoming a bit tortured :-/  It would be worth pulling out the test into a local `isAutoManaging()` method that will use `useLocalSdk` to determine if auto-managing.